src/client/datascience/jupyter/kernels/jupyterKernelService.ts

```diff
@@ -260,6 +269,13 @@ export class JupyterKernelService {
                     return;
                 }
 
+                // Ensure we inherit env variables from the original kernelspec file.
+                const envInKernelSpecJson =
+                    getKernelRegistrationInfo(kernel) === 'registeredByNewVersionOfExtForCustomKernelSpec'
+                        ? specModel.env || {}
+                        : {};
+                specModel.env = Object.assign(envInKernelSpecJson, specModel.env);
```

Comment:
When re-generating env variables, we need to preserve the env variables the user may have created in their custom kernelspec files.